src/fu-engine-requirements.c

```diff
@@ -210,30 +300,18 @@ fu_engine_requirements_check_firmware(FuEngine *self,
 
 	/* check fwupd version requirement */
 	if (depth < 0) {
-		if (fu_version_compare(fwupd_version, "1.9.7", FWUPD_VERSION_FORMAT_UNKNOWN) < 0) {
-			g_set_error(error,
-				    FWUPD_ERROR,
-				    FWUPD_ERROR_NOT_SUPPORTED,
-				    "requirement 'child firmware' also needs %s >= 1.9.7",
-				    FWUPD_DBUS_SERVICE);
+		if (!fu_engine_requirements_check_fwupd_version(helper, "1.9.7", error)) {
+			g_prefix_error(error, "requirement child firmware: ");
 			return FALSE;
 		}
 	} else if (depth == 0) {
-		if (fu_version_compare(fwupd_version, "1.6.1", FWUPD_VERSION_FORMAT_UNKNOWN) < 0) {
-			g_set_error(error,
-				    FWUPD_ERROR,
-				    FWUPD_ERROR_NOT_SUPPORTED,
-				    "requirement 'sibling firmware' also needs %s >= 1.6.1",
-				    FWUPD_DBUS_SERVICE);
+		if (!fu_engine_requirements_check_fwupd_version(helper, "1.6.1", error)) {
+			g_prefix_error(error, "requirement sibling firmware: ");
 			return FALSE;
 		}
 	} else if (depth == 1) {
-		if (fu_version_compare(fwupd_version, "1.3.4", FWUPD_VERSION_FORMAT_UNKNOWN) < 0) {
-			g_set_error(error,
-				    FWUPD_ERROR,
-				    FWUPD_ERROR_NOT_SUPPORTED,
-				    "requirement 'parent firmware' also needs %s >= 1.3.4",
-				    FWUPD_DBUS_SERVICE);
+		if (!fu_engine_requirements_check_fwupd_version(helper, "1.3.4", error)) {
```

Comment:
kinda dead code at this point no?

Reply:
I'm not sure; people are *using* super old fwupd versions, so I still think it makes sense to check the requirements are set correctly -- it also enforces to the uploaders that the fwupd version is something they should care about -- and forces them to choose a "lowest possible" version when uploading.

Reply:
But we're talking about fwupd daemon code.  Can't we already assume that we don't care about the requirements because we have a new enough version?

Those "checks" are really for helping craft a cab more than anything I guess.

Reply:
Yup, although the checks are not needed here -- I still think they're useful exactly for the `build-cabinet` case were someone builds the archive on a new fwupd, which then gets deployed (wrongly) on an old fwupd.

Reply:
I'm thinking maybe you're right; we rip out all the version checks and we learn more heavily into the feature flag thing -- it works better for the way we're backporting the important stuff into 1.9.x. Maybe We should split out the feature flags into some new bitfield rather than special casing the "daemon feature" thing.